MetaMask/metamask-extension
# Ability to buy tokens with Wyre

## Explanation

This extends this already merged PRs: https://github.com/MetaMask/metamask-extension/pull/15551  https://github.com/MetaMask/metamask-extension/pull/15924
and adds Wyre as an extra onramp provider for tokens

Current State: Only native currencies can be bought through Wyre using the buy button
After Change: Ability to purchase tokens defined in `BUYABLE_CHAINS_MAP` using Wyre



## More Information

This depends on a swap-api changed which has already been deployed, this is why it hasn't been included on previous PRs



## Screenshots/Screencaps



### Before
<img width="356" alt="Screenshot 2022-09-27 at 12 03 01" src="https://user-images.githubusercontent.com/8658960/192498257-4d1872cb-e0c6-46a5-9e90-e1503ffb326d.png">



### After
<img width="357" alt="Screenshot 2022-09-27 at 12 02 31" src="https://user-images.githubusercontent.com/8658960/192498311-c355efc0-6ea3-4824-9e8f-40b2789b4f02.png">




## Manual Testing Steps

Import a token
If token is on the Wyre lists it should display a "Buy" button
Once you click on the provider's "Continue to" button, a window on said provider should open, and the token to buy should already be preselected



## Pre-Merge Checklist


### + If there are functional changes:
